ONSdigital/takeon-business-layer
# Feature/spp1950/da

# Description
This pull request covers the Business Layer changes to handle and display messages for DA in UI. The changes involved are

- Created Enumeration interface which has key value pairs for Error Flag and the corresponding Error descriptions.
- Modified the existing GraphQL query to join DateAdjustmentContributor table to retrieve Error Flag information for a given contributor
- Parsed GraphQL query to provide Error Flag and corresponding Error description in UI for display
- Added unit tests and refactored the code. 

# How to test?

- Search a contributor for a given RSI survey
- Click the View Form Button
- Enter some form values and click Save and Validate Button
- You can see a error flag and the corresponding error description in UI for any major errors for Date Adjustment (Form Status will be 'Form saved'

# Unit Tests
You can run java unit tests using: mvn clean test

# Code Coverage
You can verify code coverage using: awk -F"," '{ instructions += $4 + $5; covered += $5 } END { print covered, "/", instructions, " instructions covered"; print 100*covered/instructions, "% covered" }' target/site/jacoco/jacoco.csv

# Check Style errors
You can verify check style errors using : mvn package install

# Links
https://collaborate2.ons.gov.uk/jira/browse/SPP-1950
